crates/router/src/utils/user/dashboard_metadata.rs

```diff
@@ -279,9 +279,15 @@ pub fn parse_string_to_enums(query: String) -> UserResult<GetMultipleMetaDataPay
     })
 }
 
-pub fn is_prod_email_required(data: &ProdIntent) -> bool {
-    !(data
-        .poc_email
+fn not_contains_string(value: &Option<String>, value_to_be_checked: &str) -> bool {
+    value
         .as_ref()
-        .map_or(true, |mail| mail.contains("juspay")))
+        .map_or(false, |mail| !mail.contains(value_to_be_checked))
+}
+
+pub fn is_prod_email_required(data: &ProdIntent, user_email: String) -> bool {
+    not_contains_string(&data.poc_email, "juspay")
+        && not_contains_string(&data.business_website, "juspay")
+        && not_contains_string(&data.business_website, "hyperswitch")
+        && not_contains_string(&Some(user_email), "juspay")
 }
```

Comment:
lets use [is_some_and](https://doc.rust-lang.org/std/option/enum.Option.html#method.is_some_and) when we update rust version.